### PR TITLE
Update to very latest version of SmallRye RM and MP api

### PR DESCRIPTION
**Pull request by tevans78**
#build

### CODE REVIEW

File: dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.reactive.messaging-1.0.feature [hunk 1]
```diff
@@ -2,5 +2,8 @@
 symbolicName=com.ibm.websphere.appserver.org.eclipse.microprofile.reactive.messaging-1.0
 singleton=true
 -bundles=com.ibm.websphere.org.eclipse.microprofile.reactive.messaging.1.0; location:="dev/api/stable/,lib/"; mavenCoordinates="org.eclipse.microprofile.reactive.messaging:microprofile-reactive-messaging-api:1.0"
+-features=\
```
Comment (tevans78) [inline dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.reactive.messaging-1.0.feature]: this bit might have been a mistake? will fix tomorrow

Comment (tevans78) [reply dev/com.ibm.websphere.appserver.features/visibility/private/com.ibm.websphere.appserver.org.eclipse.microprofile.reactive.messaging-1.0.feature]: decided this is actually ok

**Comment by Azquelt** [comment]
Need to expose new package `org.eclipse.microprofile.reactive.messaging.spi` as stable API from the reactive messaging feature.

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java [hunk 2]
```diff
@@ -40,10 +40,11 @@ void beforeBeanDiscovery(@Observes BeforeBeanDiscovery discovery, BeanManager be
...
         addAnnotatedType(MediatorManager.class, discovery, beanManager);
         addAnnotatedType(StreamProducer.class, discovery, beanManager);
-        addAnnotatedType(ConfiguredStreamFactory.class, discovery, beanManager);
-        addAnnotatedType(InternalStreamRegistry.class, discovery, beanManager);
-        addAnnotatedType(StreamFactoryImpl.class, discovery, beanManager);
         addAnnotatedType(MediatorFactory.class, discovery, beanManager);
+        addAnnotatedType(InternalChannelRegistry.class, discovery, beanManager);
+        addAnnotatedType(ConfiguredChannelFactory.class, discovery, beanManager);
+        addAnnotatedType(LegacyConfiguredChannelFactory.class, discovery, beanManager);
+
```
Comment (Azquelt) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: Also need
```
addQualifier(Connector.class, discovery, beanManager);
```

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java [hunk 3]
```diff
@@ -29,9 +29,9 @@
...
 import io.smallrye.reactive.messaging.extension.ReactiveMessagingExtension;
 import io.smallrye.reactive.messaging.extension.StreamProducer;
-import io.smallrye.reactive.messaging.impl.ConfiguredStreamFactory;
-import io.smallrye.reactive.messaging.impl.InternalStreamRegistry;
-import io.smallrye.reactive.messaging.impl.StreamFactoryImpl;
+import io.smallrye.reactive.messaging.impl.ConfiguredChannelFactory;
+import io.smallrye.reactive.messaging.impl.InternalChannelRegistry;
+import io.smallrye.reactive.messaging.impl.LegacyConfiguredChannelFactory;
 
 @Component(service = WebSphereCDIExtension.class, configurationPolicy = ConfigurationPolicy.IGNORE, property = { "service.vendor=IBM" })
```
Comment (Azquelt) [inline dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: Need to add property `"application.bdas.visible=true"`

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java (hunk 2 above)
Comment (tevans78) [reply dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: done

File: dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java (hunk 3 above)
Comment (tevans78) [reply dev/com.ibm.ws.io.smallrye.reactive.messaging/src/com/ibm/ws/microprofile/reactive/messaging/OLReactiveMessagingExtension.java]: done

**Comment by tevans78** [comment]
#build

**Pull request merged**